src/client/App.jsx

```diff
@@ -1,56 +1,119 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import { Routes, Route, Link, useNavigate } from "react-router-dom";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
-  const [movies, setMovies] = useState([]);
+  const [movies, setMovies] = useState(null);
+  const [updateMovieList, setUpdateMovieList] = useState(0);
+
+  const navigate = useNavigate();
 
   useEffect(() => {
-    fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
-  }, []);
+    fetch(`${apiUrl}/movie`, {
+      headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+    })
+      .then((res) => res.json())
+      .then((res) => setMovies(res.data));
+  }, [updateMovieList]);
 
   const handleRegister = async ({ username, password }) => {
-    
+    const url = "http://localhost:4000/user/register";
+    const opts = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ username, password }),
+    };
+
+    const fetchRes = await fetch(url, opts);
```

Comment:
Choose one convention - either  async / await or promise syntax

Reply:
Everything has the async / await syntax ✔️